tests/datalab/issue_manager/test_utils.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+import pytest
+
+# from scipy.sparse import csr_matrix
+
+from cleanlab.datalab.internal.issue_manager.utils import ConstructedKNNGraph
+from cleanlab.datalab.datalab import Datalab
+
+
+class TestConstructedKNNGraph:
```

Comment:
Here is the same class with some suggestions for the test cases. You can check if this covers all the relevant cases. 
```suggestion
from scipy.sparse import csr_array, csr_matrix

class TestConstructedKNNGraph:

    @pytest.fixture
    def sparse_matrix_csr(self,):
        return csr_array((5, 5))

    @pytest.fixture
    def constructed_knn_graph_instance(self, lab):
        return ConstructedKNNGraph(lab)

    def test_process_knn_graph_from_inputs_vaid_graph(self, constructed_knn_graph_instance, sparse_matrix_csr):

        # Check when knn_graph is present in kwargs
        sparse_matrix_csr = csr_array((5, 5)).toarray()
        kwargs = {"knn_graph": sparse_matrix_csr}
        knn_graph = constructed_knn_graph_instance.process_knn_graph_from_inputs(kwargs)
        assert isinstance(knn_graph, csr_matrix) # Assert type
        assert knn_graph is sparse_matrix_csr # Assert that passed sparse matrix is same as returned knn graph

        # Check when knn_graph is present in "statistics"
        lab = constructed_knn_graph_instance.datalab
        lab.info["statistics"]["weighted_knn_graph"] = sparse_matrix_csr # Set key in statistics
        knn_graph = constructed_knn_graph_instance.process_knn_graph_from_inputs(kwargs={})
        assert isinstance(knn_graph, csr_matrix) # Assert type
        assert knn_graph is sparse_matrix_csr

    # Any other cases where valid knn_graph is returned


    def test_process_knn_graph_from_inputs_return_None(self, constructed_knn_graph_instance, sparse_matrix_csr):

        # First check for no knn graph (not present in kwargs or "weighted_knn_graph")
        kwargs = {}
        knn_graph = constructed_knn_graph_instance.process_knn_graph_from_inputs(kwargs)
        assert knn_graph is None
        # Pass knn_graph in CSC format
        kwargs = {"knn_graph": sparse_matrix_csr.tocsc()}
        knn_graph = constructed_knn_graph_instance.process_knn_graph_from_inputs(kwargs)
        assert isinstance(knn_graph, csr_matrix) # Assert type
        assert knn_graph is None # Assert that passed sparse matrix is same as returned knn graph

        # Any other cases where returned knn_graph is None
```